sklearn/preprocessing/data.py

```diff
@@ -2369,16 +2389,47 @@ def _transform_col(self, X_col, quantiles, inverse):
 
         return X_col
 
+    def _assert_finite_or_nan(self, X):
+        """Check that X contain finite or NaN values."""
+        X = np.asanyarray(X)
+        # First try an O(n) time, O(1) space solution for the common case that
+        # everything is finite; fall back to O(n) space np.isfinite to prevent
+        # false positives from overflow in sum method.
+        if (X.dtype.char in np.typecodes['AllFloat']
+                and not np.isfinite(X[~np.isnan(X)].sum())
+                and not np.isfinite(X[~np.isnan(X)]).all()):
+            raise ValueError("Input contains infinity"
+                             " or a value too large for %r." % X.dtype)
+        if np.count_nonzero(self._mask_missing_values):
```

Comment:
It is strange to have this in a function whose argument is X, not _mask_missing_values.

Reply:
I suspect you should avoid storing this mask as an attribute.

Reply:
This function will get away once the #10455 is addressed.